rasa/nlu/utils/spacy_utils.py

```diff
@@ -88,21 +83,82 @@ def cache_key(
         cls, component_meta: Dict[Text, Any], model_metadata: "Metadata"
     ) -> Optional[Text]:
 
-        # Fallback, use the language name, e.g. "en",
-        # as the model name if no explicit name is defined
-        spacy_model_name = component_meta.get("model", model_metadata.language)
+        spacy_model_name = cls.check_model_fallback(
+            component_meta.get("model"), model_metadata.language, warn=True
+        )
 
         return cls.name + "-" + spacy_model_name
 
+    @classmethod
+    def check_model_fallback(cls, spacy_model_name, language_name, warn=False):
+        """This method checks if the `spacy_model_name` is missing and attempts to do a fallback.
+
+        This feature is a measure to support spaCy 3.0 without breaking on users. In the future
+        spaCy will no longer support `spacy link`.
+        """
+        if not spacy_model_name:
+            fallback_mapping = {
+                "zh": "zh_core_web_md",
+                "da": "da_core_news_md",
+                "nl": "nl_core_news_md",
+                "en": "en_core_web_md",
+                "fr": "fr_core_news_md",
+                "de": "de_core_news_sm",
+                "el": "el_core_news_md",
+                "it": "it_core_news_md",
+                "ja": "ja_core_news_md",
+                "lt": "lt_core_news_md",
+                "mk": "mk_core_news_md",
+                "nb": "nb_core_news_md",
+                "pl": "pl_core_news_md",
+                "pt": "pt_core_news_md",
+                "ro": "ro_core_news_md",
+                "ru": "ru_core_news_md",
+                "es": "es_core_news_md",
+            }
+            if warn:
+                warnings.warn(
```

Comment:
lets please use `raise_deprecation_warning`. I'd also suggest to have only one warning instead of 4 which is raised in case of the fallback is attempted

Reply:
@koaning

Reply:
Are you referring to a function called `raise_deprecation_warning`? If so, I couldn't find one in Rasa so I assumed you are referring to raising a `DeprecationWarning` here? I've moved everything into a single warning.